container/docker/handler.go

```diff
@@ -135,18 +135,18 @@ func newDockerContainerHandler(
 	id := ContainerNameToDockerId(name)
 
 	// Add the Containers dir where the log files are stored.
-	otherStorageDir := path.Join(path.Dir(storageDriverDir), pathToContainersDir, id)
+	otherStorageDir := path.Join(storageDir, pathToContainersDir, id)
 
-	rwLayerID, err := getRwLayerID(id, storageDriverDir, dockerVersion)
+	rwLayerID, err := getRwLayerID(id, storageDir, storageDriver, dockerVersion)
 	if err != nil {
 		return nil, err
 	}
 	var rootfsStorageDir string
 	switch storageDriver {
```

Comment:
@jimmidyson: Do you have access to a zfs machine? I'm trying to add a zfs machine to the test fleet.